packages/expo-updates/android/src/main/java/expo/modules/updates/db/DatabaseIntegrityCheck.java

```diff
@@ -0,0 +1,49 @@
+package expo.modules.updates.db;
+
+import androidx.annotation.Nullable;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import expo.modules.updates.db.entity.AssetEntity;
+import expo.modules.updates.db.entity.UpdateEntity;
+
+public class DatabaseIntegrityCheck {
+  public void run(UpdatesDatabase database, File updatesDirectory, @Nullable UpdateEntity embeddedUpdate) {
+    List<AssetEntity> assets = database.assetDao().loadAllAssets();
+
+    ArrayList<AssetEntity> missingAssets = new ArrayList<>();
+    for (AssetEntity asset : assets) {
+      if (asset.relativePath == null) {
```

Comment:
if `relative_path` is null in the database, marking the asset as missing again is a no-op -- is it intentional to call `missingAssets.add(asset);` below?

Reply:
I believe their status is updated as well: https://github.com/expo/expo/pull/12607/files#diff-9ea333a02417c08aa0ab35b0ca99eb9ca3d97112f061c0a42d6d1ad6eb029446R60